src/Storage/PluginManager.php

```diff
@@ -14,57 +19,57 @@
  * Plugin\PluginInterface. Additionally, it registers a number of default
  * plugins available.
  *
- * @extends AbstractPluginManager<PluginInterface>
+ * @extends AbstractSingleInstancePluginManager<PluginInterface>
+ * @psalm-import-type ServiceManagerConfiguration from ServiceManager
  */
-final class PluginManager extends AbstractPluginManager
+final class PluginManager extends AbstractSingleInstancePluginManager
 {
-    /** @var array<string,string> */
-    protected $aliases = [
-        'clear_expired_by_factor' => Plugin\ClearExpiredByFactor::class,
-        'clearexpiredbyfactor'    => Plugin\ClearExpiredByFactor::class,
-        'clearExpiredByFactor'    => Plugin\ClearExpiredByFactor::class,
-        'ClearExpiredByFactor'    => Plugin\ClearExpiredByFactor::class,
-        'exception_handler'       => Plugin\ExceptionHandler::class,
-        'exceptionhandler'        => Plugin\ExceptionHandler::class,
-        'exceptionHandler'        => Plugin\ExceptionHandler::class,
-        'ExceptionHandler'        => Plugin\ExceptionHandler::class,
-        'ignore_user_abort'       => Plugin\IgnoreUserAbort::class,
-        'ignoreuserabort'         => Plugin\IgnoreUserAbort::class,
-        'ignoreUserAbort'         => Plugin\IgnoreUserAbort::class,
-        'IgnoreUserAbort'         => Plugin\IgnoreUserAbort::class,
-        'optimize_by_factor'      => Plugin\OptimizeByFactor::class,
-        'optimizebyfactor'        => Plugin\OptimizeByFactor::class,
-        'optimizeByFactor'        => Plugin\OptimizeByFactor::class,
-        'OptimizeByFactor'        => Plugin\OptimizeByFactor::class,
-        'serializer'              => Plugin\Serializer::class,
-        'Serializer'              => Plugin\Serializer::class,
-    ];
-
-    /** @var array<string,string> */
-    protected $factories = [
-        Plugin\ClearExpiredByFactor::class => InvokableFactory::class,
-        Plugin\ExceptionHandler::class     => InvokableFactory::class,
-        Plugin\IgnoreUserAbort::class      => InvokableFactory::class,
-        Plugin\OptimizeByFactor::class     => InvokableFactory::class,
-        Plugin\Serializer::class           => InvokableFactory::class,
+    private const CONFIGURATION = [
+        'aliases'   => [
+            'clear_expired_by_factor' => Plugin\ClearExpiredByFactor::class,
+            'clearexpiredbyfactor'    => Plugin\ClearExpiredByFactor::class,
+            'clearExpiredByFactor'    => Plugin\ClearExpiredByFactor::class,
+            'ClearExpiredByFactor'    => Plugin\ClearExpiredByFactor::class,
+            'exception_handler'       => Plugin\ExceptionHandler::class,
```

Comment:
Is there any value in keeping all of these aliases alive? Would it not be a good time to get rid of them and force users to reference the required implementation by FQCN?

Reply:
Just BC

Reply:
I was also thinkin of this but besides the fact that it would provide a benefit to projects with psalm, I think its okayish to keep them. Might be something to consider in next major but I would do that with a better way to configure adapters in total, not just enforcing FQCN here.